Makefile

```diff
@@ -99,15 +99,6 @@ $(cmds):
 
 build: $(cmds)
 
-deploy-operator: operator generate
```

Comment:
Do we have these instructions anywhere else? They may be useful for developers.

Reply:
I agree, having it either in the readme or as a build-target in the Makefile doesn't hurt. So far the issue IMO has been that a target named "deploy-operator" does a lot more than deploying. One needs the generator-tool and go 1.15 to deploy? I was rather surprised.

Reply:
`cmd/operator/README` has documentation too and in fact was slightly conflicting with the top-level README and `make deploy-operator`. With this change the idea is that users deploy the operator with `kubectl`.

I'm OK to keep `Makefile` as it is if it helps development. @rojkov thoughts?

Reply:
These targets are a copy-paste from the kubebuilder book. I used them for operator development and for end users they have zero value. But after I had added webhooks to the operator these targets became rather worthless.

Reply:
Probably we can keep them, but with a comment that they are useful for developing operators without webhooks.

Reply:
I'd suggest we drop these. `make envtest` is still available for quick local sanity checks

Reply:
I don't see developer instructions in cmd/operator/README. I agree that we can drop them from the top level README.md, but we should have them somewhere.